pkg/k8shandler/indexmanagement/index_management.go

```diff
@@ -0,0 +1,170 @@
+package indexmanagement
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	loggingv1alpha1 "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
+	esapi "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+)
+
+const (
+	PolicyNameApp   = "app-policy"
+	PolicyNameInfra = "infra-policy"
+	PolicyNameAudit = "audit-policy"
+
+	MappingNameApp   = "node.app"
+	MappingNameInfra = "node.infra"
+	MappingNameAudit = "node.audit"
+
+	PollInterval = "12h"
+
+	HotPhaseAgeAsPercentOfMaxAge = 5
+)
+
+var (
+	AliasesApp   = []string{"app.logs"}
+	AliasesInfra = []string{"infra", "infra.node"}
+	AliasesAudit = []string{"infra.audit"}
+
+	agePattern = regexp.MustCompile("^(?P<number>\\d+)(?P<unit>[yMwdhHms])$")
+)
+
+func NewCRSpec(retentionPolicy *logging.RetentionPolicySpec) *esapi.IndexManagementSpec {
+
+	if retentionPolicy == nil {
+		return nil
+	}
+
+	indexManagement := esapi.IndexManagementSpec{}
+	if maxAgeApp, ok := retentionPolicy.MaxAge[loggingv1alpha1.LogSourceTypeApp]; ok {
```

Comment:
i think with the requested change to the struct, the next several blocks are convertable into a for loop, you might have to define additional maps to get aliases and other bits